plain_html.py

```diff
@@ -168,8 +168,11 @@ def consolidate_text(soup):
     for element in soup.find_all(string=True):
         # If the previous element is the same type then extract the current string and append to previous
         if type(element.previous_sibling) is type(element):
-            # Join with no spaces if this is a smart quotation mark
-            if str(element.previous_sibling)[-1] in ['“', '‘'] or str(element)[0] in ['”', '’']:
+            # Join with no spaces if the previous character is a opening smart quotation mark or bracket
+            join_before = ('“', '‘', '(', '[', '{')
```

Comment:
Should we also include non-smart double and single quotes or is it too tricky to figure out if they are opening or closing a quote in order to determine if we should join without whitespace before or after?

Reply:
Yes, it is the ambiguity about whether they're opening or closing quotes that makes this difficult. Maybe there's a better way to deal with this whole issue though?

Reply:
We talked yesterday about the option of not stripping / re-adding trailing and leading whitespace when joining NavigableStrings. If we do this, then we can avoid all these rules entirely.